src/constants.cpp

```diff
@@ -1909,6 +1910,8 @@ struct spell_types spells[] =
     { "Detect Magic", FALSE, DETECTION, AREA, -1, SUSTAINED, 0, LIGHT },
     { "Detect Object", TRUE, DETECTION, AREA, -1, SUSTAINED, 1, MODERATE },
     { "Mindlink", FALSE, DETECTION, SINGLE, -1, SUSTAINED, 0, SERIOUS },
+    { "Nightvision", TRUE, DETECTION, SINGLE, -1, SUSTAINED, 1, MODERATE },
```

Comment:
These have to be in the exact order they're written in awake.h, so Nightvision has to be after Splash etc.